src/index.js.flow

```diff
@@ -113,7 +113,7 @@ declare export function createChannel(connection: AmqpConnection|Promise<AmqpCon
 
 declare export function createManagedChannel(connection: AmqpManagedConnection): PublishChannel
 
-declare export function consumeFrom<C:ConsumeChannel> (channel: C): (config: ConsumeConfig, messageHandler: MessageHandler) => Promise<{}>
+declare export function consumeFrom<C:DuplexChannel> (channel: C): (config: ConsumeConfig, messageHandler: MessageHandler) => Promise<{}>
```

Comment:
I wonder why flow didn't complain when I did this originally. :/

My *intention* was to make DuplexChannel a subtype of ConsumeChannel, such that a function that accepts a ConsumeChannel can also accept a DuplexChannel.  However, there's a bit of extra complexity 😢 in how I had to do it, which is why the AmqpConsumeChannel, Channel<AmqpConsumeChannel>, AmqpDuplexChannel and Channel<AmqpDuplexChannel> type aliases exist.  I'd love to find a way to simplify that, but my haskell-type-system brain couldn't figure out a better way in flow.

Anyway, it seems like the problem is that, even tho AmqpDuplexChannel *is* a compatible subtype of AmqpConsumeChannel, flow *doesn't* consider Channel<AmqpDuplexChannel> to be a subtype of Channel<AmqpConsumeChannel>, or to put it in the nerdiest possible terms, the flow type Channel<A> isn't covariant.

I wonder if flow's [variance modifiers](http://flowtype.org/docs/classes.html#polymorphism-and-type-parameter-variance) can help here.  I vaguely remember playing around with that, and not having any luck, but that was probably several flow versions ago.

I'm ok with this change in the short term so this PR isn't a blocker, but I'd like to find a better solution at some point!  Maybe just add a FIXME or TODO to figure out if there's a way to make the subtyping work.